notes/cryptanalysis.tex

```diff
@@ -2,33 +2,200 @@
 \section{Block Cipher Design and Cryptanalysis}
 \label{sec:cryptanalysis}
 
-We've seen some theoretical ways to build block ciphers, namely Feistel
-constructions using round functions that are secure as PRFs. There are other
-constructions as well, such as Evan-Mansour \tnote{Should probably include
-somewhere}. But this is reductive, since it's unclear how to build the PRFs
-themselves. One could imagine trying to use actual random functions, but this is
-intractable for large block sizes --- the secret key in this case being a random
-table requring $n2^n$ bits. 
+So far we've seen some theoretical ways to construct block ciphers, namely Feistel networks using round functions that are as secure are PRFs. There are other ways to build block ciphers such as the Even-Mansour~\cite{EvenMansour} construction from a single known PRP. It is basically of the form: 
+\[ E_{<K_1, K_2>}(M) = F(M \xor K_1) \xor K_2 \] 
+Here, \( K_1, K_2 \) are the keys used for Message \(M\) and \(F\) is a PRP that is known (or can be easily obtained) for the Even-Mansour encryption scheme \(E\). 
+\par But these kinds of designs can prove to be reductive since the mechanisms to build PRFs in practice is itself unclear. One could try to use actual random functions. But this is untractable for large block sizes,  the secret key, in this case, being a random table requiring $n2^n$ bits
+(For block sizes of n, the lookup table has to have at least $2^n$ possible n-bit string values to look indistinguishable from a random function for
+a particular key). \newline
```

Comment:
formatting comment: use blank lines to separate paragraphs instead of `\newline`